test/wcennz.test.ts

```diff
@@ -0,0 +1,111 @@
+import { expect, use } from 'chai';
+import { Contract, utils } from 'ethers';
+import { deployContract, MockProvider, solidity } from 'ethereum-waffle';
+import CENNZnetBridge from '../artifacts/contracts/CENNZnetBridge.sol/CENNZnetBridge.json';
+import ERC20Peg from '../artifacts/contracts/ERC20Peg.sol/ERC20Peg.json';
+import MockBridge from '../artifacts/contracts/MockBridge.sol/MockBridge.json';
+import WrappedCENNZ from '../artifacts/contracts/WrappedCENNZ.sol/WrappedCENNZ.json';
+
+use(solidity);
+
+describe('Erc20Peg', () => {
+    const [wallet, walletTo] = new MockProvider().getWallets();
+    let bridge: Contract;
+    let erc20Peg: Contract;
+    let mockBridge: Contract;
+    let wrappedCENNZ: Contract;
+
+    beforeEach(async () => {
+        bridge = await deployContract(wallet, CENNZnetBridge, []);
+        mockBridge = await deployContract(wallet, MockBridge, []);
+        erc20Peg = await deployContract(wallet, ERC20Peg, [mockBridge.address]);
+        wrappedCENNZ = await deployContract(wallet, WrappedCENNZ, [erc20Peg.address]);
+    });
+
+    it('withdraw wrapped CENNZ then deposit', async () => {
+        let withdrawalAmount = 10;
+        let userBalanceStart = await wrappedCENNZ.balanceOf(wallet.address);
+        await erc20Peg.activateWithdrawals();
+        await wrappedCENNZ.approve(erc20Peg.address, withdrawalAmount);
+
+        // SETUP: bridge contract validators
+        let validatorPublicKey = '0x0204dad6fc9c291c68498de501c6d6d17bfe28aee69cfbf71b2cc849caafcb0159';
+        let validatorAddress = utils.computeAddress(validatorPublicKey);
+        let validatorSetId = 0;
+        await bridge.forceActiveValidatorSet(
+            // 'Alice' default CENNZnet ECDSA public key converted to Eth address
+            [validatorAddress],
+            validatorSetId,
+        );
+        let verificationFee = await bridge.verificationFee();
+        // A CENNZnet validator signature for withdraw event: (ETH_RESERVED_TOKEN_ADDRESS, 5644, 0xa86e122EdbDcBA4bF24a2Abf89F5C230b37DF49d)
+        let signature = utils.splitSignature(
+            utils.hexlify('0x67bb4327409a32ce8acd415a836ba9106d3371281cc8dc646f075c18a21717701b48bd8b57c4193a20acef0c3ab7a6fcc5f7f6b27e199b1a51b38bda857bdc1601')
+        );
+        let withdrawProof = {
+            eventId: 1,
+            validatorSetId: 0,
+            validators: [validatorAddress],
+            v: [signature.v],
+            r: [signature.r],
+            s: [signature.s],
+        };
```

Comment:
nit: would do something like this so it's obvious its fake (also unneeded setup stuff)
```suggestion
        let fakeWithdrawProof = {
            eventId: 0,
            validatorSetId: 0,
            validators: [],
            v: [],
            r: [],
            s: [],
        };
```